plugins/enterkey/plugin.js

```diff
@@ -405,6 +414,42 @@
 
 			range.select();
 			range.scrollIntoView();
+
+			// ===== HELPERS =====
+			function preventExtraLineInsideTable( mode ) {
+				// #1816
+				// We want to have behaviour after pressing enter like this:
+				// 1. <td>^</td> -> <td><p> </p></td>
+				// 2. <td>Foo^</td> -> <td><p>Foo</p><p> </p></td>
+				// 3. <td>Foo^Bar</td> -> <td><p>Foo</p><p>Bar</p></td>
+				// We need to separate 1. case to not add extra line. Like it happen for 2nd or 3rd option.
+
+				var innerElement,
+					bogus;
+
+				if ( mode === CKEDITOR.ENTER_BR ) {
+					return false;
+				}
+
+				if ( CKEDITOR.tools.indexOf( [ 'td', 'th' ], path.lastElement.getName() ) === -1 ) {
+					return false;
+				}
+				if ( path.lastElement.getChildCount() !== 1 ) {
+					return false;
+				}
+
+				innerElement = path.lastElement.getChild( 0 ).clone( true );
```

Comment:
Is this cloning logic really needed? I've tested in Firefox and IE 11 with following code:

```javascript
var element = CKEDITOR.dom.element.createFromHtml( '<div><br/></div>' );
element.getBogus(); // <br/>
element.getText(); // ''
element.getText().length; // 0
```

Seems like browsers (at least Firefox & IE 11) treat bogus element as non-existent.

Reply:
At least in chrome it is problematic, because paragraph might be removed in such situation, and cell became unelectable. That's why using cloned element seems to be more proper here.
![apr-30-2018 10-52-28](https://user-images.githubusercontent.com/9017322/39420433-daefbac6-4c64-11e8-95f6-b1572e80ff03.gif)